PubHub/PubHub.Common/Services/ServiceInstanceResult.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PubHub.Common.Services
+{
+    public class ServiceInstanceResult<T>
```

Comment:
You could let this one inherit from `ServiceResult` and reuse the `StatusCode` and `ErrorDescriptor` properties.

`public class ServiceInstanceResult<T> : ServiceResult`